services/src/main/java/org/apache/unomi/services/impl/rules/RulesServiceImpl.java

```diff
@@ -335,6 +365,7 @@ public Rule getRule(String ruleId) {
         Rule rule = persistenceService.load(ruleId, Rule.class);
         if (rule != null) {
             ParserHelper.resolveConditionType(definitionsService, rule.getCondition(), "rule " + rule.getItemId());
+            updateRulesByEventType(rulesByEventType, rule);
```

Comment:
Yes you are right I will remove it. However it should not duplicate the rule since we use Sets and Rule inherits from Item that defines the equals() and hashcode() methods to be used with sets.